mod-configuration-server/src/main/java/org/folio/rest/impl/ConfigAPI.java

```diff
@@ -371,32 +402,32 @@ public void getConfigurationsAudit(String query, int offset,
                   List<Audit> auditList = (List<Audit>) reply.result().getResults();
                   auditRecords.setAudits(auditList);
                   auditRecords.setTotalRecords(reply.result().getResultInfo().getTotalRecords());
-                  asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetConfigurationsAuditResponse.withJsonOK(
+                  asyncResultHandler.handle(succeededFuture(GetConfigurationsAuditResponse.withJsonOK(
                     auditRecords)));
                 }
                 else{
                   log.error(reply.cause().getMessage(), reply.cause());
-                  asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetConfigurationsAuditResponse
+                  asyncResultHandler.handle(succeededFuture(GetConfigurationsAuditResponse
                     .withPlainInternalServerError(messages.getMessage(
                       lang, MessageConsts.InternalServerError))));
                 }
               } catch (Exception e) {
                 log.error(e.getMessage(), e);
-                asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetConfigurationsAuditResponse
+                asyncResultHandler.handle(succeededFuture(GetConfigurationsAuditResponse
                   .withPlainInternalServerError(messages.getMessage(
                     lang, MessageConsts.InternalServerError))));
               }
             });
       }
       catch(CQLQueryValidationException e1){
```

Comment:
log.error(e1.getMessage()) is missing from this catch block.
This catch block is duplicated code, see other CQLQueryValidationException catch block above. Can you extract the code into some function?

Reply:
@julianladisch Which catch block are you referring to? When I look at line 416, it appeas to have a log.error in place.

In general, I have tried to make minimal changes to the production code with this change. There are lots of aspects that could be cleaned up and I didn't want to get too bogged down in them within this issue (it was already much bigger than intended).

Do you think this is a necessary change for this pull request, or something to be done at some point?

Reply:
@julianladisch Ah, do you mean the one starting line 422 now?

If so, I've added the logging and extracted a function.

Reply:
Also, apologies for my initially defensive response, this change has been rather frustrating at times, I should not have taken that out on you :-(

Thank you for taking the time to review this change thoroughly and for the feedback, it has been useful.